Symfony/CS/Fixer/Contrib/EregToPregFixer.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+/*
+ * This file is part of the Symfony CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Matteo Beccati <matteo@beccati.com>
+ */
+class EregToPregFixer extends AbstractFixer
+{
+    /**
+     * @var array
+     */
+    private $map = array(
+        array('ereg', 'preg_match', ''),
+        array('eregi', 'preg_match', 'i'),
+        array('ereg_replace', 'preg_replace', ''),
+        array('eregi_replace', 'preg_replace', 'i'),
+        array('split', 'preg_split', ''),
+        array('spliti', 'preg_split', 'i'),
+    );
+
+    /**
+     * @var array
+     */
+    private $delimiters = array('/', '#', '!');
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        if (!$this->cursoryMatch($content)) {
+            return $content;
+        }
+
+        $tokens = Tokens::fromCode($content);
+
+        foreach ($this->map as $map) {
+            $seq = array(array(T_STRING, $map[0]), '(');
+            $currIndex = 0;
+            while (null !== ($ereg = $tokens->findSequence($seq, $currIndex, $tokens->count(), false))) {
+                // Ensure it's a function call
+                $prev = $tokens->getPrevMeaningfulToken($ereg[0]);
+                if (!isset($prev) || $tokens[$prev]->isGivenKind(array(T_OBJECT_OPERATOR, T_DOUBLE_COLON))) {
+                    break;
+                }
+
+                // Extract the string to convert
+                $index = $tokens->getNextMeaningfulToken($ereg[1]);
+                if (!isset($index) || !$tokens[$index]->isGivenKind(T_CONSTANT_ENCAPSED_STRING)) {
+                    break;
+                }
+
+                // Ensure the string is not appended something or worse
+                $next = $tokens->getNextMeaningfulToken($index);
+                if (!isset($next) || !in_array($tokens[$next]->getContent(), array(',', ')'))) {
+                    break;
+                }
+
+                // Advance tokenizer cursor
+                $currIndex = $index;
+
+                // Convert to PCRE
+                $string = substr($tokens[$index]->getContent(), 1, -1);
+                $quote = substr($tokens[$index]->getContent(), 0, 1);
+                $delim = $this->getBestDelimiter($string);
```

Comment:
@sstok So, I rushed to make the change... Actually, I think it would make things much worse, as it would require parsing the ereg to avoid issues:

```php
var_dump(ereg("}foo", "}foobar"));
```

is perfectly legal and it would break when converted, generating a broken preg: "{}foo}D".